processor/filterprocessor/config.go

```diff
@@ -39,8 +48,30 @@ type Config struct {
 	Logs LogFilters `mapstructure:"logs"`
 
 	Spans SpanFilters `mapstructure:"spans"`
+
+	Conditions Conditions `mapstructure:"conditions"`
+}
+
+type Conditions struct {
+	Resource  string `mapstructure:"resource"`
+	Scope     string `mapstructure:"scope"`
+	Span      string `mapstructure:"span"`
+	SpanEvent string `mapstructure:"spanevent"`
+	Metric    string `mapstructure:"metric"`
+	DataPoint string `mapstructure:"datapoint"`
+	Log       string `mapstructure:"log"`
```

Comment:
I think we should not offer resource and scope, since it is not clear what dropping a "resource" mean. e.g. does it mean dropping the resource or the resource and all the data associated with it? 

Also would start with only supporting logrecord/metric/span initially, then we can extend.